content/en/docs/ops/ambient/helm-installation/index.md

```diff
@@ -0,0 +1,90 @@
+---
+title: Installing Ambient Mesh with Helm
+description: How to install Ambient Mesh with Helm.
+weight: 4
+owner: istio/wg-environments-maintainers
+test: n/a
+---
+
+This guide shows you how to install Ambient Mesh with Helm.
+Besides the demo in [Getting Started with Ambient Mesh](/docs/ops/ambient/getting-started/),
+we **encourage** you to follow this guide to install Ambient Mesh.
+Helm helps you manage components separately, and you can easily upgrade the components to the latest version.
+
+## Setup Repo Info
+
+{{< text bash >}}
+$ helm repo add istio https://istio-release.storage.googleapis.com/charts
+$ helm repo update
+{{< /text >}}
+
+*See [helm repo](https://helm.sh/docs/helm/helm_repo/) for command documentation.*
+
+## Installing the Components
+
+To make it easier to install the components, we have created some Helm value files for you. Make sure you have
+pulled the Helm chart package, or download the [latest version of Istio](/docs/setup/getting-started/#download).
+There will be `ambient-values.yaml` files in the chart package.
+
+In the following steps, we will use the downloaded Istio release package as an example, and make sure you are in the
+root directory of the Istio release package.
+
+### Installing Base Component
+
+The **Base** chart contains the basic CRDs and cluster roles required to set up Istio.
+This should be installed prior to any other Istio component.
+
+{{< text bash >}}
+$ helm install istio-base istio/base
+{{< /text >}}
+
+### Installing CNI Component
+
+The **CNI** chart installs the Istio CNI Plugin. It is responsible for detecting the pods that belong to the ambient mesh,
+and configuring the traffic redirection between the ztunnels - which will be installed later.
+
+{{< text bash >}}
+$ helm install istio-cni istio/cni -n kube-system \
+  -f manifests/charts/istio-cni/ambient-values.yaml
```

Comment:
This will only be for those that have downloaded the charts. If you are using the Helm repo you'll need to give the full URL to the ambient values file, such as `-f https://raw.githubusercontent.com/istio/istio/master/manifests/charts/istio-cni/ambient-values.yaml`

Reply:
Can also use the Istio.io code import: https://preliminary.istio.io/latest/docs/releases/contribute/code-blocks/#import-code-from-an-external-source-through-a-url and render (that version of the file) inline in the doc for copypasting, like so, which might be a little more obvious:


```
helm install istio-cni istio/cni -n kube-system -f - << EOF
cni:
  # Configuration log level of istio-cni binary
  # by default istio-cni send all logs to UDS server
  # if want to see them you need change global.logging.level with cni:debug
  logLevel: info

  # Allow the istio-cni container to run in privileged mode, needed for some platforms (e.g. OpenShift)
  privileged: true

  # Configure ambient settings
  ambient:
    # If enabled, ambient redirection will be enabled
    enabled: true

  # Default excludes istio-system; its actually fine to redirect there since we opt-out istiod, ztunnel, and istio-cni
  excludeNamespaces:
  - kube-system
EOF
```

Reply:
TBQH if base case requires a large values file for default/demo install, that's a good signal we have too many values - for instance in the above example it's not clear that `privileged` couldn't simply be implicitly set to true if `ambient.enabled` is true, and dropped from the values file.

Reply:
> Can also use the Istio.io code import: https://preliminary.istio.io/latest/docs/releases/contribute/code-blocks/#import-code-from-an-external-source-through-a-url and render (that version of the file) inline in the doc for copypasting, like so, which might be a little more obvious

I'm going to try this to see if it works, thanks for the info!

Reply:
> TBQH if base case requires a large values file for default/demo install, that's a good signal we have too many values with unreasonable defaults - for instance in the above example it's not clear that privileged couldn't simply be implicitly set to true if ambient.enabled is true, and dropped from the ambient values file.

I think this makes the components easier to install, but it may potentially lack flexibility because users may want to change a subset of values under the 'grouped' `ambient.enabled`, rather than the entire set. Additionally, having separate values can help indicate the differences between the original mode and the ambient mode, which is useful for users who are migrating from the sidecar mode to ambient. Given this, IMO if Ambient is defined as a separate default configuration, it might be better to simplify the Helm install, and otherwise it may be better to keep the current settings.

Reply:
Looks like we cannot have embeded texts...
<img width="595" alt="image" src="https://github.com/istio/istio.io/assets/20888172/8704b3bb-a4e4-4a84-b070-940be27597e1">
Maybe just revise to https://preliminary.istio.io/latest/docs/releases/contribute/code-blocks/#link-2-files?

Reply:
> > TBQH if base case requires a large values file for default/demo install, that's a good signal we have too many values with unreasonable defaults - for instance in the above example it's not clear that privileged couldn't simply be implicitly set to true if ambient.enabled is true, and dropped from the ambient values file.
> 
> I think this makes the components easier to install, but it may potentially lack flexibility because users may want to change a subset of values under the 'grouped' `ambient.enabled`, rather than the entire set. Additionally, having separate values can help indicate the differences between the original mode and the ambient mode, which is useful for users who are migrating from the sidecar mode to ambient. Given this, IMO if Ambient is defined as a separate default configuration, it might be better to simplify the Helm install, and otherwise it may be better to keep the current settings.

If they have a legitimate need to do that we can add that later. It's much easier to expose something as an option after the fact when someone makes a case for it then it is to un-expose when it's clear no one really needs to individually set it.

We should expose the minimal set of things as `values` that are required to install Istio across the platforms and environments we support - everything beyond that, people can make a case for in a PR.

In the case of `privileged` here for the CNI there's no real case for us to let that be individually toggleable outside of the `ambient` flag - it has to be true for the `ambient` feature to work, and even if we change that as an implementation detail, there's no reason not to make it implicit, since users changing it to anything else if they set the other flag would be an explicit misconfiguration.

Reply:
Thanks for clarifying this, makes sense.